# Bonus: Temperature Analysis I

In [44]:
# Import dependencies
import pandas as pd
import matplotlib as plt
from sqlalchemy import create_engine, func,inspect
from sqlalchemy.orm import Session
from sqlalchemy.ext.automap import automap_base
from scipy import stats
from numpy import mean
from sqlalchemy import and_

In [16]:
# Using either SQLAlchemy or pandas, get the requested data per the README
path="sqlite:///Resources/hawaii.sqlite"
engine=create_engine(path)

In [43]:
#reflect 
Base=automap_base()
Base.prepare(engine,reflect=True)
#get the classes
Base.classes.keys()


['measurement', 'station']

In [42]:
Measurement=Base.classes.measurement
Station=Base.classes.station

In [18]:
#opening the session and defining the classes
session=Session(engine)

In [23]:
#Identify the average temperature in June at all stations across all available years in the dataset.
#Do the same for December temperature.
june_temp=session.query(func.avg(Measurement.tobs).filter(func.strftime("%m", Measurement.date) == "06").all()
december_temp=session.query(func.avg(Measurement.tobs)).filter(func.strftime("%m", Measurement.date) == "12").all()
print(june_temp,december_temp)

[(74.94411764705882,)] [(71.04152933421226,)]


In [35]:
#Identify the average temperature in June at all stations across all available years in the dataset.
#Do the same for December temperature.
june_temp=session.query(Measurement.tobs).filter(func.strftime("%m", Measurement.date) == "06").all()
december_temp=session.query(Measurement.tobs).filter(func.strftime("%m", Measurement.date) == "12").all()


In [39]:
#gettin into seperate lists
june_temp_list=[]
for temp in june_temp:
    june_temp_list.append(temp.tobs)
dec_temp_list=[]
for temp in december_temp:
    dec_temp_list.append(temp.tobs)    

Here I will like to use paired ttest (same samples group from two different time)


In [41]:
# Use t-test to determine if difference in means is statistically significant
stats.ttest_ind(june_temp_list,dec_temp_list)
#stats.ttest_rel(june_temp_list,dec_temp_list)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

# Discuss which t-test to use and why and what your results mean.
i think we shd use the unpaired ttest here.
pvalue =3.9.we are accepting the null and rejecting the alternate hypothesis as the values are greater than .05. this means we are in status quo, there is not much diff in june and dec temp of hawaii.
the mean values june=74.94411764705882, july =71.04152933421226 confirm my findings.
